packages/ChangesReporting/ValueObjectFactory/ConsoleFileDiffFactory.php

```diff
@@ -0,0 +1,32 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Rector\ChangesReporting\ValueObjectFactory;
+
+use Rector\Core\Console\Formatter\ConsoleDiffer;
+use Rector\Core\FileSystem\FilePathHelper;
+use Rector\Core\ValueObject\Application\File;
+use Rector\Core\ValueObject\Reporting\FileDiff;
+
+final class ConsoleFileDiffFactory
+{
+    public function __construct(
+        private readonly ConsoleDiffer $consoleDiffer,
+        private readonly FilePathHelper $filePathHelper,
+    ) {
+    }
+
+    public function createFileDiff(File $file, string $oldContent, string $newContent): FileDiff
+    {
+        $relativeFilePath = $this->filePathHelper->relativePath($file->getFilePath());
+
+        // always keep the most recent diff
+        return new FileDiff(
+            $relativeFilePath,
+            '', // computing the diff can be slow, therefore we only compute what we need
```

Comment:
this file is pretty similar to the already existing `FileDiffFactory`.
main difference is, that we hardcoded the empty-string here, instead of computing the default-diff

Reply:
looking at it again, I think we sometimes need the "default" diff and sometimes the "console"-diff.
it could make sense to rework it so we only built the one required.

I will stop investing time here and wait for a initial round of feedback